.github/workflows/go-function-offline-testing.yml

```diff
@@ -0,0 +1,51 @@
+---
+name: golang-function-testing
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  go-example:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        tests: [
+          { directory: 'cors-go', go-version: '1.19' },
+          { directory: 'go-mnq-sqs-publish', go-version: '1.18' },
+          { directory: 'go-upload-file-s3-multipart', go-version: '1.20' }
+        ]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Go ${{ matrix.tests.go-version }}
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.tests.go-version}}
+
+      - name: Run test for ${{ matrix.tests.directory }}
+        working-directory: functions/${{ matrix.tests.directory }}
+        run: |
+          go run test/main.go &
```

Comment:
test/main.go missing ?
btw you should consider using a TestMain() function ? (not sure if it's the best case for you case)

Reply:
I'm not sure I understand what you mean by test/main.go missing ? 
In each golang example, there is a folder with `test/main.go` file that uses the `serverless-functions-go` package to be able to test the function locally. Can you explain a bit more what you have in mind?